pkg/local-gadget-manager/local-gadget-manager.go

```diff
@@ -286,6 +288,31 @@ func (l *LocalGadgetManager) Dump() string {
 	return out
 }
 
+// ensureBPFMount ensures /sys/fs/bpf is of type bpf. It is necessary to be able
+// to pin eBPF maps. TODO: Remove the need of using pinning, see issues #619 and
+// #620.
+func ensureBPFMount() error {
+	const bpfPinPath string = "/sys/fs/bpf"
+	var buf unix.Statfs_t
+
+	if err := unix.Statfs(bpfPinPath, &buf); err != nil {
+		return fmt.Errorf("error checking type of %s: %w", bpfPinPath, err)
+	}
+
+	if buf.Type != unix.BPF_FS_MAGIC {
+		log.Debugf("%s is of type %#x but we were expecting %#x (BPF_FS_MAGIC)",
+			bpfPinPath, buf.Type, unix.BPF_FS_MAGIC)
+
+		log.Infof("Remounting %q as bpf", bpfPinPath)
+
+		if err := unix.Mount("bpf", bpfPinPath, "bpf", 0, ""); err != nil {
+			return fmt.Errorf("error remounting %s as bpf: %w", bpfPinPath, err)
+		}
+	}
```

Comment:
nit: usually doing an early return simplifies this kind of code:

```

if buf.Type == unix.BPF_FS_MAGIC {
  // All good, nothing to do! 
  return nil
}

// logic to remount it. 
```

Reply:
I wasn't using this format because that code was inside `NewManager()` but now that we moved it out as an independent function, I agree it should follow the early return approach. Thanks!